tests/unit/utils/test_utils_record.py

```diff
@@ -294,3 +300,9 @@ def test_get_value_returns_none_on_index_error():
     })
 
     assert get_value(single_title, 'titles.title[1]') is None
+
+
+def test_get_record_type_returns_none_when_no_type():
+    record = Record({})
+
+    assert get_record_type(record) is None
```

Comment:
No need to overcomplicate this: just put a `$schema` value of `'does-not-exist.json'` and document what your implementation is doing in this case.